src/d.ts

```diff
@@ -36,6 +36,28 @@ export function isHNode(child: DNode): child is HNode {
 	return Boolean(child && (typeof child !== 'string') && child.type === HNODE);
 }
 
+/**
+ * Generic decorate function for DNodes, the modified in place based on the provided predicate
+ * and modifier functions.
+ *
+ * If no predicate is supplied then the modifier will be executed on all nodes.
+ */
+export function decorate(dNodes: DNode | DNode[], modifier: (dNode: DNode) => void, predicate?: (dNode: DNode) => void): DNode | DNode[] {
+	let nodes = Array.isArray(dNodes) ? [ ...dNodes ] : [ dNodes ];
+	while (nodes.length) {
+		const node = nodes.pop();
+		if (node) {
+			if (!predicate || predicate(node)) {
```

Comment:
Ok. I guess the name predicate was throwing me off... in Dojo 1.x and in other APIs I'm familiar with, it would have been called filter probably. Either way, this is fine.